sigmastate/src/test/scala/sigmastate/serialization/DeserializationResilience.scala

```diff
@@ -255,6 +288,12 @@ class DeserializationResilience extends SerializationSpecification
       ValueSerializer.deserialize(reader(ValueSerializer.serialize(expr), maxTreeDepth = 3))
   }
 
+  property("max recursive call depth is checked in writer.level for TypeSerializer") {
+    val expr = Tuple(Tuple(IntConstant(1)))
+    an[SerializeCallDepthExceeded] should be thrownBy
+      ValueSerializer.serialize(expr, writer(maxTreeDepth = 2))
```

Comment:
Same issues as with the previous test I think.

Reply:
Correct, the TypeSerializer was completely missing handling for maxTreeDepth for both de/serializing, added to both method and updated tests.